Sources/Confidence/EventSenderEngine.swift

```diff
@@ -54,19 +55,30 @@ final class EventSenderEngineImpl: EventSenderEngine {
         uploadReqChannel.sink { [weak self] _ in
             do {
                 guard let self = self else { return }
-                try self.storage.startNewBatch()
-                let ids = try storage.batchReadyIds()
-                for id in ids {
-                    let events: [NetworkEvent] = try self.storage.eventsFrom(id: id)
-                        .compactMap { event in
-                            return NetworkEvent(
-                                eventDefinition: event.name,
-                                payload: NetworkStruct(fields: TypeMapper.convert(structure: event.payload).fields),
-                                eventTime: Date.backport.toISOString(date: event.eventTime))
+                var batches: [[ConfidenceEvent]] = []
+                try queue.sync {
+                    try self.storage.startNewBatch()
+                    let ids = try storage.batchReadyIds()
+                    for id in ids {
+                        let events: [ConfidenceEvent] = try self.storage.eventsFrom(id: id)
+                        batches.append(events)
+                        do {
+                            try storage.remove(id: id)
```

Comment:
If the app crashes before these events are processed we are going to lose them

Reply:
yes, that's the compromise to not make the uploads sync (blocking, one after another).